### PR TITLE
Core, some worlds: Rename sweep_for_events to sweep_for_advancements

### PR DESCRIPTION
**Pull request by NewSoupVi**
Seems like we're going with the "advancement" naming now, and I have recent evidence that people think this function is referring to `address = None`. So let's try to get away from the old "event" naming and be consistent about "event" meaning `address = None` and "advancement" meaning "will be picked up by sweep_for_events / etc."

I wasn't sure about the `event` parameter on `collect`. I feel like that one *might* be used slightly differently? Lmk what you think about that one.

### CODE REVIEW

**Comment by NewSoupVi** [comment]
@Alchav

**Comment by beauxq** [comment]
no deprecated `sweep_for_events` to not break apworlds?

**Comment by NewSoupVi** [comment]
> no deprecated `sweep_for_events` to not break apworlds?

My bad, I had that, and then I accidentally removed it again lol

**Comment by NewSoupVi** [comment]
Approval from Espeon is implicit https://discord.com/channels/731205301247803413/1214608557077700720/1253206955879694336

**Review by Jouramie — APPROVED** [review]
I confirm that this was super confusing when first getting into AP. Good change 👍

**Comment by Alchav** [comment]
> @Alchav (Can't request review)

What, do you really need my approval to rename a core function just because I use it?

**Comment by NewSoupVi** [comment]
> > Alchav (Can't request review)
> 
> What, do you really need my approval to rename a core function just because I use it?

That is the current procedure, yes - If something touches a world, the world maintainer has to approve.

We have discussed changing this in the (recent) past, some world maintainers have argued that we shouldn't need to do this, but other world maintainers have said they want to sign off on anything touching their world.

Happy to discuss solutions in the Discord.

File: worlds/alttp/test/dungeons/TestDungeon.py
Comment (agilbert1412) [inline worlds/alttp/test/dungeons/TestDungeon.py:58]: That comment used to be col-aligned with the one above, and now it's not anymore.
Personally I don't think aligning comments has much value, and I would instead squeeze it closer to the code on its line, but regardless of preference, the current placement seems wrong

**Review by agilbert1412 — APPROVED** [review]
1: Approved, that seems like a good thing
2: Why was I requested to review this?
3: Note for the future, if something requires approval from Stardew, even though technically it's my name on it, I consider if Jouramie approves it, it's good enough, as he wrote most of that world with me and is arguably even more qualified than me to approve things related to it

**Review by Alchav — APPROVED** [review]

File: worlds/alttp/test/dungeons/TestDungeon.py
Comment (NewSoupVi) [reply worlds/alttp/test/dungeons/TestDungeon.py:58]: Answer for 2: It touched Stardew in a previous iteration, then I force-pushed over that, but still pinged you not realizing I no longer need to. It's my b, sorry <3

**Review by Alchav — APPROVED** [review]

**Comment by NewSoupVi** [comment]
Oh thanks for the reapproval Alchav, did I accidentally ping you again? If so, I didn't mean to :D

File: BaseClasses.py
Comment (alwaysintreble) [inline BaseClasses.py:759]: This should probably use Utils.deprecate

Comment (NewSoupVi) [reply BaseClasses.py:759]: Tru

**Review by hatkirby — APPROVED** [review]
Oh, I thought this PR was some weird dream I had, but it turns out it's real.

**Comment by NewSoupVi** [comment]
Oh Violet, extremely smart and beautiful world maintainer of the highly acclaimed The Witness Randomizer for AP, are you willing to acquiesce to this change?

**Comment by NewSoupVi** [comment]
Yes, esteemed and prolific core maintainer NewSoupVi, this change is beautiful and you should be proud of yourself, hats off to you my friend

**Comment by NewSoupVi** [comment]
Thank you lovely Violet, please continue on with your day, may it be an awesome one like you deserve!

File: BaseClasses.py
Comment (Berserker66) [inline BaseClasses.py:760]:
```suggestion
        return self.sweep_for_advancements(locations)
```
Not needed, as it has no return, just "more correct".

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**
